packages/std/src/math/signed_decimal.rs

```diff
@@ -0,0 +1,3060 @@
+        match self.checked_mul(other) {
+            Ok(value) => value,
+            Err(_) => {
+                // both negative or both positive results in positive number, otherwise negative
+                if self.is_negative() == other.is_negative() {
+                    Self::MAX
+                } else {
+                    Self::MIN
+                }
+            }
+        }
+    }
+
+    #[must_use = "this returns the result of the operation, without modifying the original"]
+    pub fn saturating_pow(self, exp: u32) -> Self {
+        match self.checked_pow(exp) {
+            Ok(value) => value,
+            Err(_) => {
+                // odd exponent of negative number results in negative number
+                // everything else results in positive number
+                if self.is_negative() && exp % 2 == 1 {
+                    Self::MIN
+                } else {
+                    Self::MAX
+                }
+            }
+        }
+    }
+
+    /// Converts this decimal to a signed integer by rounding down
+    /// to the next integer, e.g. 22.5 becomes 22 and -1.2 becomes -2.
+    ///
+    /// ## Examples
+    ///
+    /// ```
+    /// use core::str::FromStr;
+    /// use cosmwasm_std::{SignedDecimal, Int128};
+    ///
+    /// let d = SignedDecimal::from_str("12.345").unwrap();
+    /// assert_eq!(d.to_int_floor(), Int128::new(12));
+    ///
+    /// let d = SignedDecimal::from_str("-12.999").unwrap();
+    /// assert_eq!(d.to_int_floor(), Int128::new(-13));
+    ///
+    /// let d = SignedDecimal::from_str("-0.05").unwrap();
+    /// assert_eq!(d.to_int_floor(), Int128::new(-1));
+    /// ```
+    #[must_use]
+    pub fn to_int_floor(self) -> Int128 {
+        if self.is_negative() {
+            // Using `x.to_int_floor() = -(-x).to_int_ceil()` for a negative `x`,
+            // but avoiding overflow by implementing the formula from `to_int_ceil` directly.
+            let x = self.0;
+            let y = Self::DECIMAL_FRACTIONAL;
+            if x.is_zero() {
+                Int128::zero()
+            } else {
+                // making sure not to negate `x`, as this would overflow
+                -Int128::one() - ((-Int128::one() - x) / y)
+            }
+        } else {
+            self.to_int_trunc()
+        }
+    }
+
+    /// Converts this decimal to a signed integer by truncating
+    /// the fractional part, e.g. 22.5 becomes 22.
+    ///
+    /// ## Examples
+    ///
+    /// ```
+    /// use core::str::FromStr;
+    /// use cosmwasm_std::{SignedDecimal, Int128};
+    ///
+    /// let d = SignedDecimal::from_str("12.345").unwrap();
+    /// assert_eq!(d.to_int_trunc(), Int128::new(12));
+    ///
+    /// let d = SignedDecimal::from_str("-12.999").unwrap();
+    /// assert_eq!(d.to_int_trunc(), Int128::new(-12));
+    ///
+    /// let d = SignedDecimal::from_str("75.0").unwrap();
+    /// assert_eq!(d.to_int_trunc(), Int128::new(75));
+    /// ```
+    #[must_use]
```

Comment:
Maybe: #[must_use = "this returns the result of the operation, without modifying the original"]

Reply:
I think I will add it for the `to_int_*` functions because those really are a bit ambiguous.